app/controllers/v1/listings_controller.rb

```diff
@@ -1,14 +1,29 @@
 class V1::ListingsController < ApplicationController
+
   def index
     page = params[:page]
     page ||= 1
 
-    @listings = Listing.page(page).per(30)
+    @listings = Listing.includes([:listing_videos, :listing_images, :seller])
+                  .page(page).per(30)
 
     render_success(status: :ok, data: serialized_listing)
   end
 
   def serialized_listing
     ListingSerializer.new(@listings || @listing).serializable_hash[:data]
   end
+
+
+  def show
+    @listing = Listing.friendly.find(params[:id])
+
+    if @listing
+      render_success(status: ok, data: ListingSerializer.new(@listing))
```

Comment:
When you call `render_success`, `status: "ok` is a default value, so you don't need to explicitly call it with status: :ok, you can omit that, and pass status: only when status is not :ok.
Also, please use `serialized_listing` insteed of passing `data: ListingSerializer.new(@listing)`, just call `data: serlized_listing`